botocove/cove_host_account.py

```diff
@@ -128,128 +125,106 @@ def _get_boto3_sts_client(self, assuming_session: Optional[Session]) -> STSClien
 
-        for current_id in ids:
-            if not isinstance(current_id, str):
-                raise TypeError(
-                    f"{current_id} is an incorrect type: all account and ou id's must be strings not {type(current_id)}"  # noqa E501
-                )
-            if re.match(r"^\d{12}$", current_id):
-                accounts.append(current_id)
-                continue
-            if re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", current_id):
-                ous.append(current_id)
-                continue
-            raise ValueError(
-                f"provided id is neither an aws account nor an ou: {current_id}"
+        if self.provided_ignore_ids is not None:
+            ignored_accounts = self._list_accounts_for_all_resources(
+                self.provided_ignore_ids
             )
+        else:
+            ignored_accounts = set()
+
+        target_accounts = included_accounts - ignored_accounts - {self.host_account_id}
+
+        return {
+            _id: account
+            for _id, account in account_map.items()
+            if _id in target_accounts
+        }
+
+    def _list_accounts_for_all_resources(self, resource_ids: List[str]) -> Set[str]:
+        accounts: Set[str] = set()
+        for res in resource_ids:
+            accounts.update(self._list_accounts_for_resource(res))
+        return accounts
+
+    def _list_accounts_for_resource(self, resource_id: str) -> Set[str]:
+        if self._is_account_id(resource_id):
+            return {resource_id}
+        if self._is_organizational_unit_id(resource_id):
+            return self._list_descendant_accounts(resource_id)
+        raise ValueError(
+            f"provided id is neither an aws account nor an ou: {resource_id}"
+        )
+
+    def _is_organizational_unit_id(self, resource_id: str) -> bool:
+        return bool(re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", resource_id))
 
-        return accounts, ous
+    def _is_account_id(self, resource_id: str) -> bool:
+        return bool(re.match(r"^\d{12}$", resource_id))
 
-    def _get_all_accounts_by_organization_units(
-        self, target_ous: List[str]
-    ) -> List[str]:
+    def _list_descendant_accounts(self, ancestor_id: str) -> Set[str]:
+        descendant_accounts = self._list_accounts_for_parent(ancestor_id)
 
-        account_list: List[str] = []
+        child_ous = self._list_organizational_units_for_parent(ancestor_id)
 
-        for parent_ou in target_ous:
+        for ou in child_ous:
+            descendant_accounts.update(self._list_descendant_accounts(ou))
 
-            current_ou_list: List[str] = []
+        return descendant_accounts
 
-            # current_ou_list is mutated and recursivly populated with all childs
-            self._get_all_child_ous(parent_ou, current_ou_list)
+    @lru_cache
+    def _list_accounts_for_parent(self, parent_id: str) -> Set[str]:
+        pages = self.org_client.get_paginator("list_accounts_for_parent").paginate(
```

Comment:
I think a try/except block just to specifically log "you're probably not running in an org master" just to guide the failure case of non-org master passing an OU id would be a nice addition - this looks to be the place it'd surface